src/topology/bornology/basic.lean

```diff
@@ -0,0 +1,142 @@
+/-
+Copyright (c) 2022 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import order.filter.cofinite
+
+/-!
+# Basic theory of bornology
+
+We develop the basic theory of bornologies. Instead of axiomatizing bounded sets and defining
+bornologies in terms of those, we recognize that the cobounded sets form a filter and define a
+bornology as a filter of cobounded sets which contains the cofinite filter.  This allows us to make
+use of the extensive library for filters, but we also provide the relevant connecting results for
+bounded sets.
+-/
+
+open set filter
+
+variables {α β : Type*}
+
+
+/-- A **bornology** on a type `α` is a filter of cobounded sets which contains the cofinite filter.
+Such spaces are equivalently specified by their bounded sets, see `bornology.of_bounded`
+and `bornology.eq_iff_is_bounded`-/
+@[protect_proj]
+structure bornology (α : Type*) :=
+(cobounded : filter α)
+(le_cofinite : cobounded ≤ cofinite)
+
+attribute [class] bornology
```

Comment:
Would it work making this be `class bornology`? Are there places where it's better to need to include an explicit `bornology` argument?  It seems to me it would be better instead to set it up so these functions take an explicit `α` argument.  By the way, you can use the `ext` attribute to generate the extensionality lemma (called `bornology.ext` rather than your `bornology.eq`).

```suggestion
@[ext]
class bornology (α : Type*) :=
(cobounded [] : filter α)
(le_cofinite [] : cobounded ≤ cofinite) -- I'm not sure if the `[]` here is necessary in practice.
```

Here's a gist showing how it might work: https://gist.github.com/kmill/9c62f9fbdaaabe27e1b0640463ff16fd

Reply:
I tried `(le_cofinite : cobounded ≤ cofinite)` without the `[]`, and the lemma that used `le_cofinite` was still able to be elaborated fine, so for now it seems justifiable leaving it off.

Reply:
Then, if you define `is_cobounded`, the bornology can be inferred from the set.  Like `is_open` and `is_closed`, we could use those rather than mentioning the `cobounded` filter directly:
```lean
def is_cobounded (s : set α) : Prop := s ∈ cobounded α

lemma is_cobounded_def {s : set α} : is_cobounded s ↔ s ∈ cobounded α := iff.rfl

lemma is_bounded_compl_iff {s : set α} : is_bounded sᶜ ↔ is_cobounded s :=
by rw [is_bounded_def, is_cobounded_def, compl_compl]
```

Reply:
Thanks. In all honesty, it was a bit of cargo cult coding based on `topology.basic`. I had assumed that elaboration would have problems but I didn't check.